test/e2e_node/pod_hostnamefqdn_test.go

```diff
@@ -71,11 +71,7 @@ func testPod(podnamebase string) *v1.Pod {
 
 var _ = SIGDescribe("Hostname of Pod [Feature:SetHostnameAsFQDN][NodeFeature:SetHostnameAsFQDN]", func() {
 	f := framework.NewDefaultFramework("hostfqdn")
-	dnsDomain := "cluster.local"
-	if cdn := framework.TestContext.ClusterDNSDomain; cdn != "" {
-		dnsDomain = cdn
-	}
-
+	dnsDomain := framework.TestContext.ClusterDNSDomain
```

Comment:
The default value has been set in test/e2e/framework/test_context.go in RegisterClusterFlags, right?

Reply:
Hmm, not sure. The issue I see is that cluster domain is "", but the default should be cluster.local (see https://github.com/kubernetes/kubernetes/blob/512b7b75a8da19e669e058bf13d5978560c4c65f/test/e2e/framework/test_context.go#L334). Maybe I am doing something wrong in my e2e config or maybe framework.NewDefaultFramework sets dns-domain to "", I cannot find it though.

Reply:
I add dns-domain to the kubelet-flags when testing, but framework.TestContext.ClusterDNSDomain remains "", even though Pod now have cluster.local

`make test-e2e-node FOCUS="\[NodeFeature:SetHostnameAsFQDN\]" PARALLELISM=1 TEST_ARGS='--kubelet-flags="--fail-swap-on=false" --kubelet-flags="--cluster-domain=cluster.local"'
`
Maybe I should follow up in a new issue, current change should help Node CI to stop failing these tests.

Reply:
I think I found the issue with test-e2e-node, I will submit another PR suggesting a change that will make test-e2e-node similar to what hack/ginkgo-e2e.sh does.

Reply:
I created https://github.com/kubernetes/kubernetes/pull/96315 to have cluster.local as default DNS cluster-domain